src/ImageSharp/Formats/Webp/Lossy/Vp8Encoding.cs

```diff
@@ -68,51 +81,299 @@ static Vp8Encoding()
+                InverseTransformHorizontalPass(t0, t2, t1, t3, out Vector128<short> shifted0, out Vector128<short> shifted1, out Vector128<short> shifted2, out Vector128<short> shifted3);
+
+                // Transpose the two 4x4.
+                LossyUtils.Vp8Transpose_2_4x4_16b(shifted0, shifted1, shifted2, shifted3, out t0, out t1, out t2, out t3);
+
+                // Add inverse transform to 'ref' and store.
+                // Load the reference(s).
+                Vector128<byte> ref0 = Vector128<byte>.Zero;
+                Vector128<byte> ref1 = Vector128<byte>.Zero;
+                Vector128<byte> ref2 = Vector128<byte>.Zero;
+                Vector128<byte> ref3 = Vector128<byte>.Zero;
+                ref byte referenceRef = ref MemoryMarshal.GetReference(reference);
+
+                // Load eight bytes/pixels per line.
+                ref0 = Vector128.Create(Unsafe.As<byte, long>(ref referenceRef), 0).AsByte();
+                ref1 = Vector128.Create(Unsafe.As<byte, long>(ref Unsafe.Add(ref referenceRef, WebpConstants.Bps)), 0).AsByte();
+                ref2 = Vector128.Create(Unsafe.As<byte, long>(ref Unsafe.Add(ref referenceRef, WebpConstants.Bps * 2)), 0).AsByte();
+                ref3 = Vector128.Create(Unsafe.As<byte, long>(ref Unsafe.Add(ref referenceRef, WebpConstants.Bps * 3)), 0).AsByte();
+
+                // Convert to 16b.
+                ref0 = Sse2.UnpackLow(ref0, Vector128<byte>.Zero);
+                ref1 = Sse2.UnpackLow(ref1, Vector128<byte>.Zero);
+                ref2 = Sse2.UnpackLow(ref2, Vector128<byte>.Zero);
+                ref3 = Sse2.UnpackLow(ref3, Vector128<byte>.Zero);
+
+                // Add the inverse transform(s).
+                Vector128<short> ref0InvAdded = Sse2.Add(ref0.AsInt16(), t0.AsInt16());
+                Vector128<short> ref1InvAdded = Sse2.Add(ref1.AsInt16(), t1.AsInt16());
+                Vector128<short> ref2InvAdded = Sse2.Add(ref2.AsInt16(), t2.AsInt16());
+                Vector128<short> ref3InvAdded = Sse2.Add(ref3.AsInt16(), t3.AsInt16());
+
+                // Unsigned saturate to 8b.
+                ref0 = Sse2.PackUnsignedSaturate(ref0InvAdded, ref0InvAdded);
+                ref1 = Sse2.PackUnsignedSaturate(ref1InvAdded, ref1InvAdded);
+                ref2 = Sse2.PackUnsignedSaturate(ref2InvAdded, ref2InvAdded);
+                ref3 = Sse2.PackUnsignedSaturate(ref3InvAdded, ref3InvAdded);
+
+                // Unsigned saturate to 8b.
+                ref byte outputRef = ref MemoryMarshal.GetReference(dst);
+
+                // Store eight bytes/pixels per line.
+                Unsafe.As<byte, Vector64<byte>>(ref outputRef) = ref0.GetLower();
+                Unsafe.As<byte, Vector64<byte>>(ref Unsafe.Add(ref outputRef, WebpConstants.Bps)) = ref1.GetLower();
+                Unsafe.As<byte, Vector64<byte>>(ref Unsafe.Add(ref outputRef, WebpConstants.Bps * 2)) = ref2.GetLower();
```

Comment:
In the first 3 calls, it's possible and better to avoid `GetLower`. The second store will overwrite the upper 8 bit written in the first store and so on:

```suggestion
                Unsafe.As<byte, Vector128<byte>>(ref outputRef) = ref0;
                Unsafe.As<byte, Vector128<byte>>(ref Unsafe.Add(ref outputRef, WebpConstants.Bps)) = ref1;
                Unsafe.As<byte, Vector128<byte>>(ref Unsafe.Add(ref outputRef, WebpConstants.Bps * 2)) = ref2;
```

If you guarantee at the call site that `dst` has a patting of 8 bytes, you can also avoid it at the last call.

Reply:
I dont think that will work. I only want to write 8 bytes from each ref vector. 
The first write will be at position 0, the second at 32 (note the Unsafe.Add() with `WebpConstants.Bps`, WebpConstants.Bps is 32), the third at 64 and the last at 96.

Reply:
Ah haven't noticed that `Bps` is 32, nevermind then.